### PR TITLE
Kind local test cluster

### PR DESCRIPTION
**Pull request by alina-kim**
Issue #, if available:

Description of changes: This PR will cover part of #111 
```
kialina@3c22fb7dd565 aws-controllers-k8s % ./scripts/kind-build-test.sh -s ecr
Running Docker build as you requested for ecr service
ecr
🐳 Using Kubernetes kindest/node:v1.16.9@sha256:7175872357bc85847ec4b1aba46ed1d12fa054c83ac7a8a11f5c268957fd5765
🥑 Downloading the "kubectl" binary
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 47.8M  100 47.8M    0     0  7665k      0  0:00:06  0:00:06 --:--:-- 8465k
👍 Downloaded the "kubectl" binary
🥑 Downloading the "kind" binary
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   630  100   630    0     0   1730      0 --:--:-- --:--:-- --:--:--  1730
100 9771k  100 9771k    0     0  3576k      0  0:00:02  0:00:02 --:--:-- 4362k
👍 Downloaded the "kind" binary
🥑 Downloading the "helm" binary
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 12.9M  100 12.9M    0     0  4634k      0  0:00:02  0:00:02 --:--:-- 4633k
👍 Downloaded the "helm" binary
🥑 Creating k8s cluster using "kind"
👍 Created k8s cluster using "kind"
🥑 Building ecr docker image
Sending build context to Docker daemon  922.1MB
Step 1/18 : FROM golang:1.14.1 as builder
 ---> b022a9b82d88
Step 2/18 : ARG work_dir=/github.com/aws/aws-controllers-k8s
 ---> Using cache
 ---> 1983285dee33
Step 3/18 : WORKDIR $work_dir
 ---> Using cache
 ---> 11e7b1c5189e
Step 4/18 : ENV GOPROXY=direct
 ---> Using cache
 ---> f463d9e5bd47
Step 5/18 : ENV GO111MODULE=on
 ---> Using cache
 ---> 5be481148615
Step 6/18 : ENV GOARCH=amd64
 ---> Using cache
 ---> 6d7871fda51d
Step 7/18 : ENV GOOS=linux
 ---> Using cache
 ---> f60298c981db
Step 8/18 : ENV CGO_ENABLED=0
 ---> Using cache
 ---> f14204478e15
Step 9/18 : COPY go.mod go.mod
 ---> Using cache
 ---> 1dcc066e0285
Step 10/18 : COPY go.sum go.sum
 ---> Using cache
 ---> 2a3305d71f33
Step 11/18 : RUN  go mod download
 ---> Using cache
 ---> 1b3de2b1e31f
Step 12/18 : COPY . $work_dir/
 ---> 699a568c6ae7
Step 13/18 : RUN  go build -a -o $work_dir/bin/controller $work_dir/services/ecr/cmd/controller/main.go
 ---> Running in 6a00b88f5fc9
Removing intermediate container 6a00b88f5fc9
 ---> be72a9c248e6
Step 14/18 : FROM amazonlinux:2
 ---> ba2cc467a2bc
Step 15/18 : ARG work_dir=/github.com/aws/aws-controllers-k8s
 ---> Using cache
 ---> ab1b29242b77
Step 16/18 : WORKDIR /
 ---> Using cache
 ---> c7ca14d64749
Step 17/18 : COPY --from=builder $work_dir/bin/controller /bin/.
 ---> Using cache
 ---> c12e2efb8d00
Step 18/18 : ENTRYPOINT ["/bin/controller"]
 ---> Using cache
 ---> c0c3a5d62fff
Successfully built c0c3a5d62fff
Successfully tagged ecr:8f0e6d8-dirty
👍 Built the ecr docker image
🥑 Loading the images into the cluster
Image: "ecr:8f0e6d8-dirty" with ID "sha256:c0c3a5d62fff560201aeb369aac541090f474bddcbc75820346f36ea64345a34" not yet present on node "test-27fbb37c-worker", loading...
Image: "ecr:8f0e6d8-dirty" with ID "sha256:c0c3a5d62fff560201aeb369aac541090f474bddcbc75820346f36ea64345a34" not yet present on node "test-27fbb37c-control-plane", loading...
👍 Loaded image(s) into the cluster
======================================================================================================
To poke around your test manually:
export KUBECONFIG=/Users/kialina/go/src/github.com/aws/aws-controllers-k8s/scripts/..//build/tmp-test-27fbb37c/kubeconfig
export PATH=/Users/kialina/go/src/github.com/aws/aws-controllers-k8s/scripts/..//build/tmp-test-27fbb37c:$PATH
kubectl get pods -A
======================================================================================================
🥑 Deleting k8s cluster using "kind"
Deleting cluster "test-27fbb37c" ...
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/psp-privileged.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Should grant access to very few pods, i.e. kube-system system pods and possibly CNI pods
```
Comment (nithu0115) [inline scripts/psp-privileged.yaml:1]: Delete this file. It is not used

File: scripts/kind-two-node-cluster.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+  - role: control-plane
+    kubeadmConfigPatches:
+      - |
+        apiVersion: kubeadm.k8s.io/v1beta2
+        kind: ClusterConfiguration
+        metadata:
```
Comment (nithu0115) [inline scripts/kind-two-node-cluster.yaml:9]: missing name parameter here in metadata ?

File: scripts/kind-build-test.sh [hunk 3]
```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+set -eo pipefail
+OPTIND=1
+AEMM_URL="amazon-ec2-metadata-mock-service.default.svc.cluster.local"
+AEMM_VERSION="1.2.0"
+AEMM_DL_URL="https://github.com/aws/amazon-ec2-metadata-mock/releases/download/v${AEMM_VERSION}/amazon-ec2-metadata-mock-${AEMM_VERSION}.tgz"
```
Comment (jaypipes) [inline scripts/kind-build-test.sh:6]: we don't need any of the above mock metadata things.

File: scripts/kind-build-test.sh [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    exit 1
+}
+
+function get_nth_worker_pod {
+    kubectl get pods -n kube-system \
+      --selector 'app.kubernetes.io/name=aws-node-termination-handler' \
+      --field-selector="spec.nodeName=$CLUSTER_NAME-worker,status.phase=Running" \
+      --sort-by=.metadata.creationTimestamp \
+      --output jsonpath='{.items[-1].metadata.name}'
+}
```
Comment (jaypipes) [inline scripts/kind-build-test.sh:49]: The above bash function is node-termination-handler specific and not related to ACK.

File: scripts/kind-build-test.sh [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+USAGE=$(cat << 'EOM'
+  Usage: bash run.sh [-p] [-s] [-o] [-b <TEST_BASE_NAME>] [-c <CLUSTER_CONTEXT_DIR>] [-i <AWS Docker image name>] [-s] [-v K8S_VERSION]
+  Executes a test within a provisioned kubernetes cluster with NTH and IMDS pre-loaded.
```
Comment (jaypipes) [inline scripts/kind-build-test.sh:53]: needs to be changed to not say "with NTH and IMDS pre-loaded" :)

File: scripts/kind-build-test.sh [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+      ;;
+    \? )
+        echo "${USAGE}" 1>&2
+        exit
+      ;;
+  esac
+done
+
+if [ -z $TMP_DIR ]; then
+    TMP_DIR=$("${SCRIPT_PATH}"/provision-kind-cluster.sh -b "${CLUSTER_NAME_BASE}" -v "${K8S_VERSION}" "${PROVISION_CLUSTER_ARGS}")
```
Comment (jaypipes) [inline scripts/kind-build-test.sh:106]: I think eventually I want to replace `provision-kind-cluster.sh` scritpt with a Bash function `up-kind-cluster` that functions more like this: https://github.com/jaypipes/aws-controllers-k8s/blob/master/scripts/lib/cluster.sh#L14

File: scripts/kind-build-test.sh [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+echo "kubectl get pods -A"
+echo "======================================================================================================"
+
+
+### exported vars and funcs that tests can use
+export TMP_DIR
+export CLUSTER_NAME
+export AEMM_URL
+export AEMM_VERSION
+export AEMM_DL_URL
```
Comment (jaypipes) [inline scripts/kind-build-test.sh:152]: above three AEMM variables are unused (refer to the mock metadata service that we don't need)

File: scripts/provision-kind-cluster.sh [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+K8_VERSION="$K8_1_16"
+KUBECTL_VERSION=$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)
+KIND_VERSION="0.8.1"
+HELM_VERSION="3.2.4"
+
+echoerr() { echo "$@" 1>&2; }
+
+USAGE=$(cat << 'EOM'
+  Usage: provision-cluster  [-b <BASE_CLUSTER_NAME>] [-i <TEST_IDENTIFIER>] [-v K8s_VERSION] [-o]
+  Executes the spot termination integration test for the Node Termination Handler.
```
Comment (jaypipes) [inline scripts/provision-kind-cluster.sh:28]: references the Node Termination Handler...

**Pull request merged**